File: ObjectiveC/ObjectiveC/Dog.m
//
//  Dog.m
//  ObjectiveC
//
//  Created by HackerU on 07/04/2016.
//  Copyright © 2016 Ofer Raviv. All rights reserved.
//

#import "Dog.h"

@implementation Dog
{
    int age;
    double weight;
    NSString *name;
}
-(void)bark
{
    NSLog(@"waf waf %@",name);
}
-(void)setAge:(int)a
{
    age = a;
}
-(void)setWeight:(double)w
{
    weight = w;
}
-(int)age
{
    return age;
}
-(double)weight
{
    return weight;
}
-(void)setName:(NSString*)n
{
    name = n;
}

@end

File: Class/Class/Dog.swift
//
//  Dog.swift
//  Class
//
//  Created by HackerU on 03/12/2015.
//  Copyright © 2015 HackerU. All rights reserved.
//

import Foundation

class Dog {
    private var _name:String = "name"
    private var _owner:Person?
    
    var owner:Person?{
        get{
            return _owner
        }
        set{
            if owner!.age >= 18{
                _owner = newValue
            }
        }
    }
    
    func changeOwnerAge(age:Int){
        if age >= 18{
            _owner!.age = age
        }
    }
}